src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EEMethodSymbol.cs

```diff
@@ -385,7 +385,7 @@ public override ImmutableArray<Location> Locations
 
         public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
         {
-            get { throw ExceptionUtilities.Unreachable(); }
+            get { return ImmutableArray.Create(CSharpSyntaxTree.Dummy.GetRoot().GetReference()); }
```

Comment:
>CSharpSyntaxTree.Dummy.GetRoot().GetReference() [](http://example.com/codeflow?start=47&length=47)

This implementation feels somewhat surprising. We could return an empty array, Alternatively, I think we have a helper (```GetDeclaringSyntaxReferenceHelper```?) that creates result based on locations

Reply:
For example, see
```
        public sealed override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
        {
            get
            {
                return IsImplicitlyDeclared ?
                    ImmutableArray<SyntaxReference>.Empty :
                    GetDeclaringSyntaxReferenceHelper<ParameterSyntax>(_locations);
            }
        }
```

Reply:
In src\Compilers\CSharp\Portable\Symbols\Source\SourceParameterSymbol.cs